File: 14.js
function collatz (integer){
    var counter = 1
    while (integer > 1){
        counter ++;
        if (integer % 2 == 0){
            integer = integer / 2;
        }
        else {
            integer = 3 * integer + 1;
        }
    }
    return counter;
}

sequences = []
for (i = 0; i < 1000000; i++){
    sequences.push(collatz(i));
}

function arrayMax(array){
    return array.reduce(function(a, b){
        return (a > b ? a : b);
        })
}
console.log(sequences.indexOf(525));

//prints 837799

File: 14.py
def collatz(num, counter = 1):
    while num > 1:
        counter += 1
        if num % 2 == 0:
            num = num / 2
        else:
            num  = 3 * num + 1
    return counter

collatzes = [collatz(i) for i in range(1, 1000000)]
final = collatzes.index(525)
print(final)

#prints 837798